pootle/apps/pootle_translationproject/models.py

```diff
@@ -424,5 +455,144 @@ def scan_projects(sender, instance, created=False, raw=False, **kwargs):
     if not created or raw:
         return
 
-    for project in Project.objects.enabled().iterator():
-        create_translation_project(instance, project)
+    scan_translation_projects(languages=[instance.code])
+
+
+class JobWrapper():
+    """
+    Wraps RQ Job to handle it within external `watch`,
+    encapsulates work with external to RQ job params which is needed
+    because of possible race conditions
+    """
+    def __init__(self, id, connection):
+        self.id = id
+        self.func = None
+        self.language = None
+        self.project = None
+        self.options = None
+        self.depends_on = None
+        self.origin = None
+        self.timeout = None
+        self.connection = connection
+        self.job = Job(id=id, connection=self.connection)
+
+    @classmethod
+    def create(cls, func, language, project, options, connection, origin, timeout):
+        """
+        Creates object and initializes Job ID
+        """
+        job_wrapper = cls(None, connection)
+        job_wrapper.job = Job(connection=connection)
+        job_wrapper.id = job_wrapper.job.id
+        job_wrapper.func = func
+        job_wrapper.language = language
+        job_wrapper.project = project
+        job_wrapper.options = options
+        job_wrapper.connection = connection
+        job_wrapper.origin = origin
+        job_wrapper.timeout = timeout
+
+        return job_wrapper
+
+    def get_last_job_key(self):
+
+        return POOTLE_SYNC_LAST_JOB_PREFIX + \
+               "%s.%s" % (self.language.code, self.project.code)
+
+    def create_job(self, status=None, depends_on=None):
+        """
+        Creates Job object with given job ID
+        """
+        args = (self.language, self.project)
+        return Job.create(self.func, args=args, kwargs=self.options,
+                          id=self.id, connection=self.connection,
+                          depends_on=depends_on, status=status)
+
+    def save_enqueued(self, pipe):
+        """
+        Preparing job to enqueue. Works via pipeline.
+        Nothing done if WatchError happens while next `pipeline.execute()`.
+        """
+        job = self.create_job(status=JobStatus.QUEUED)
+        job.origin = self.origin
+        job.enqueued_at = utcnow()
+        if job.timeout is None:
+            job.timeout = self.timeout
+        job.save(pipeline=pipe)
+
+    def save_deferred(self, depends_on, pipe):
+        """
+        Preparing job to defer (add as dependent). Works via pipeline.
+        Nothing done if WatchError happens while next `pipeline.execute()`.
+        """
+        job = self.create_job(depends_on=depends_on, status=JobStatus.DEFERRED)
+        job.register_dependency(pipeline=pipe)
+        job.save(pipeline=pipe)
+
+        return job
+
+
+class SyncJobFunctions(object):
+    SYNC = sync_translation_project
+    UPDATE = update_translation_project
+
+
+POOTLE_SYNC_LAST_JOB_PREFIX = "pootle:sync:last:job:"
```

Comment:
Move this to be an attribute of the JobWrapper class since that's the only place it's used. Here it's confusing what it's actually for.